### PR TITLE
feat: add white theme

### PR DESCRIPTION
**Pull request by ironsoul0**
- The white theme was added;
- Possible to toggle themes;
- Preferred theme is saved into LocalStorage.

image

image

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -27,16 +30,41 @@ class App extends Component {
         this.state = {
             text: "",
             destination: "vatriume",
-            notify: false
+            notify: false,
+            whiteTheme: false
```
Comment (BekzhanKassenov) [inline src/App.jsx]: Please add trailing comma, to avoid this kind of diffs in the future.

File: src/App.jsx [hunk 2]
```diff
@@ -27,16 +30,41 @@ class App extends Component {
...
+    }
+    generateClass(className) {
+        return className + (this.state.whiteTheme ? " white" : "")
+    }
+    toggleTheme(e) {
+        e.preventDefault()
+        this.setState(prevState => {
+            document.body.style.background = prevState.whiteTheme ? "#222222" : "#ffffff"
+            const theme = localStorage.getItem("theme") || "dark"
+            localStorage.setItem("theme", theme === "dark" ? "white" : "dark")
```
Comment (BekzhanKassenov) [inline src/App.jsx]: What is localStorage gets out of sync with the toggle state? It's not super clear that they will always be in sync.

I'd rather not look at what's stored in the local storage and only write there a new state based on prevState. In that case if they get out of sync for whatever reason, next toggling will automatically fix them.

**Review by BekzhanKassenov — CHANGES_REQUESTED** [review]
Please add couple of lines to the documentation explaining how to maintain this (i.e. what should be changed in CSS and JSX when new components/functionality is added).

I left some JS style related comments, feel free to discard them.

**Comment by BekzhanKassenov** [comment]
Hi!

I hope it's not urgent to merge this :) I'm very busy this week, I'll merge and deploy next week in one go.

**Comment by ironsoul0** [comment]
> Please add couple of lines to the documentation explaining how to maintain this (i.e. what should be changed in CSS and JSX when new components/functionality is added).
> 
> I left some JS style related comments, feel free to discard them.

Thanks, I will look at this.

**Comment by ironsoul0** [comment]
> Hi!
> 
> I hope it's not urgent to merge this :) I'm very busy this week, I'll merge and deploy next week in one go.

Sure! I know you are very busy now. By the way, please let me congratulate you on your wedding.

File: README.md [hunk 3]
```diff
@@ -22,6 +22,11 @@
 * Запусти webpack (`yarn start` или `npm start`)
 * А для билда `yarn build` или `npm run build`
 
+# Белая тема (thanks to @ironsoul0!)
+Для поддержки белой темы, стоит учитывать следующие вещи:
+* При написании новых React компонентов, вы можете задавать стили для белой темы через написание стилей для `.white .yourClassName`. При этом, стили для `.yourClassName` будут применяться для темной темы. Следовательно, в JSX от вас лишь потребуется задавать prop `className={yourClassName}`.
```
Comment (BekzhanKassenov) [inline README.md:27]: Please add a comment about adding assets and switching between them.

File: src/App.jsx [hunk 4]
```diff
@@ -94,15 +93,14 @@ class App extends Component {
     }
     render() {
         return (
-            <div className={this.generateClass("App")}>
+            <div className={"App" + (this.state.whiteTheme ? " white" : "")}>
```
Comment (BekzhanKassenov) [inline src/App.jsx]: Nice!!!

**Review by BekzhanKassenov — COMMENTED** [review]
Please resolve all old comments

File: README.md (hunk 3 above)
Comment (ironsoul0) [reply README.md]: Done.

File: src/App.jsx (hunk 1 above)
Comment (ironsoul0) [reply src/App.jsx]: > Please add trailing comma, to avoid this kind of diffs in the future.

Done.

File: src/App.jsx (hunk 2 above)
Comment (ironsoul0) [reply src/App.jsx]: > What is localStorage gets out of sync with the toggle state? It's not super clear that they will always be in sync.
> 
> I'd rather not look at what's stored in the local storage and only write there a new state based on prevState. In that case if they get out of sync for whatever reason, next toggling will automatically fix them.

Done.

**Comment by ironsoul0** [comment]
> Please add couple of lines to the documentation explaining how to maintain this (i.e. what should be changed in CSS and JSX when new components/functionality is added).
> 
> I left some JS style related comments, feel free to discard them.

Done.

File: README.md (hunk 3 above)
Comment (ironsoul0) [reply README.md:27]: > Please add a comment about adding assets and switching between them.

Done.

**Comment by ironsoul0** [comment]
In my last commit, I have formatted all the code to make it look nicer. It is fully optional and I can discard this commit if you wish. 

I believe that all other things you noted so far were fixed.

**Comment by ironsoul0** [comment]
@BekzhanKassenov 😎

**Comment by BekzhanKassenov** [comment]
Deployed successfully

**Pull request merged**